### PR TITLE
Respect TypedDict from typing_extensions

### PR DESCRIPTION
**Pull request by kornicameister**
Fixes: #50

### CODE REVIEW

**Comment by kornicameister** [comment]
@ilevkivskyi before moving further here, can you explain why following code was added:

```python
PY36_TESTS = """
class TD(TypedDict):
    x: int
    y: int
class Other(dict):
    x: int
    y: int
"""

PY36 = sys.version_info[:3] >= (3, 6, 0)
if PY36:
    exec(PY36_TESTS)
else:
    TD = Other = object  # for linters
```

I wanted to write a test to provide `TypedDict` via `pytest.mark` but well...I am a bit puzzled. I am not sure if this can be removed or not.

**Comment by ilevkivskyi** [comment]
> can you explain why following code was added

This code is a syntax error on Python 3.5 and Python 2.7 (both of which are still supported by this library).

**Comment by kornicameister** [comment]
@ilevkivskyi would that make sense?

File: test_typing_inspect.py [hunk 1]
```diff
@@ -415,6 +428,8 @@ def test_get_forward_arg(self):
         fr = get_args(tp)[0]
         self.assertEqual(get_forward_arg(fr), "FRef")
         self.assertEqual(get_forward_arg(tp), None)
+        self.assertIsNot(typed_dict_keys(TDM), TDM.__annotations__)
+        self.assertIsNot(typed_dict_keys(TDE), TDE.__annotations__)
```
Comment (ilevkivskyi) [inline test_typing_inspect.py:432]: These two lines are not needed here, they are already present in the above tests.

Comment (kornicameister) [reply test_typing_inspect.py:432]: my bad :+1:

**Comment by kornicameister** [comment]
> Thanks! Just one little comment on tests, please fix it I will merge this.

Done :+1:

**Pull request merged**